src/mon/MDSMonitor.cc

```diff
@@ -2091,7 +2091,10 @@ bool MDSMonitor::drop_mds(FSMap &fsmap, mds_gid_t gid, const mds_info_t* rep_inf
 bool MDSMonitor::check_health(FSMap& fsmap, bool* propose_osdmap)
 {
   bool do_propose = false;
+  const auto mds_beacon_mon_down_grace = g_conf()->mds_beacon_mon_down_grace;
   const auto now = mono_clock::now();
+  const bool mon_down = mon.is_mon_down();
```

Comment:
i'd suggest move the definition closer to where it is used. so we have a better idea how this variable is used. in this case, it might be better to move `mon_down` down to the `// check beacon timestamps` line.